core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1158,15 +1159,18 @@ protected long findLatestFileNameGeneration(Map<String, BlobMetaData> blobs) {
                     final String file = SNAPSHOT_INDEX_PREFIX + latest;
                     logger.warn((Supplier<?>) () -> new ParameterizedMessage("failed to read index file [{}]", file), e);
                 }
+            } else if (blobKeys.isEmpty() == false) {
+                logger.debug("Could not find a readable index-N file in a non-empty shard snapshot directory [{}]",
+                    () -> blobContainer.path());
```

Comment:
you're right no lamda needed here, as the `path` is already computed and stored as an instance variable on `AbstractBlobContainer` which all implementations inherit from, so no advantage to a lamda here.  I will change it.